overlord/snapstate/snapstate.go

```diff
@@ -195,3 +199,51 @@ func Deactivate(s *state.State, snap string) (*state.TaskSet, error) {
 
 	return state.NewTaskSet(t), nil
 }
+
+// Retrieval functions
+
+func retrieveInfo(name string, si *snap.SideInfo) (*snap.Info, error) {
+	// XXX: move some of this in snap as helper?
+	snapYamlFn := filepath.Join(snap.MinimalPlaceInfo(name, si.Revision).MountDir(), "meta", "snap.yaml")
+	meta, err := ioutil.ReadFile(snapYamlFn)
+	if os.IsNotExist(err) {
+		return nil, fmt.Errorf("cannot find mounted snap %q at revision %d", name, si.Revision)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	info, err := snap.InfoFromSnapYaml(meta)
+	if err != nil {
+		return nil, err
+	}
+
+	info.SideInfo = *si
+
+	return info, nil
+}
+
+// Snap returns the information about the snap with given name and revision.
+// Works also for a mounted candidate snap in the process of being installed.
+func Snap(s *state.State, name string, revision int) (*snap.Info, error) {
```

Comment:
For making it short, can we please preserve the `Info` part instead of `Snap` (thus, `snapstate.Info`)?